src/arch/riscv/isa.cc

```diff
@@ -804,9 +810,15 @@ ISA::setMiscReg(RegIndex idx, RegVal val)
                 if (new_val.mode != AddrXlateMode::BARE &&
                     new_val.mode != AddrXlateMode::SV39)
                     new_val.mode = cur_val.mode;
+
+                // Flush TLB if satp changed as Spike does
+                if (new_val != cur_val) {
```

Comment:
Let's look a little deeper here. If the current implementation is correct (i.e., if it functionally works without flushing the TLB), then we can probably elide this. It's also possible that the original gem5 code was wrong and we could have read stale translations.